src/main/java/edu/uw/edm/docfinity/v1/controller/EmployeeController.java

```diff
@@ -111,7 +114,84 @@ public EmployeeController(PersonService personService) {
             try {
                 sc = HttpStatus.resolve(Integer.parseInt(error.getStatusCode()));
             } catch (Exception e) {
-                // retturn 500 status
+                // return 500 status
+            }
+
+            log.warn(pe.getMessage());
+
+            throw new ResponseStatusException(sc, error.getStatusDescription(), pe);
+        }
+    }
+
+    // Docfinity sends a list of values, but PWS expects a comma-separated values
+    // as a single string. As the middleman, we need to convert a list of strings
+    // to a comma-separated values as a single string. PWS returns "400 BAD REQUEST"
+    // with a message "invalid uwregid" if ANY of the regid is invalid; PWS does
+    // not provide details about the specific values that are invalid.
+    //
+    // regidList: list of Reg IDs (from DocFinity)
+    // example:
+    // regid=6D99B6F8FB52F3B5A334175D689F6136&regid=207F7468E059802FFD54DC92CFB7C590&regid=9AAC3943F778045CFADF9DE9890FD3CE
+    // Note: DocFinity uses list of strings, not comma-separated list of strings.
+    @RequestMapping(method = RequestMethod.GET, params = "regid")
+    public ResponseEntity<Page<EmployeeDTOForSearchResult>> searchByRegIdForGet(
+            @RequestParam(value = "regid") List<String> regidList,
+            @RequestParam(value = "pageSize", required = false, defaultValue = "50") Integer pageSize) {
+
+        return searchByRegId(regidList, pageSize);
+    }
+
+    // reidList: array of Reg IDs
+    // example: ["6D99B6F8FB52F3B5A334175D689F6136","207F7468E059802FFD54DC92CFB7C590"]
+    // Note: configure datasource using json and put ${regid} in the POST body
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<Page<EmployeeDTOForSearchResult>> searchByRegIdForPost(
+            @RequestBody List<String> regidList,
+            @RequestParam(value = "pageSize", required = false, defaultValue = "50") Integer pageSize) {
+
+        return searchByRegId(regidList, pageSize);
+    }
+
+    // private methods used by both GET and POST methods
+    private ResponseEntity<Page<EmployeeDTOForSearchResult>> searchByRegId(
+            List<String> regidList, Integer pageSize) {
```

Comment:
Consider consistent use of camelcase
```suggestion
            List<String> regIdList, Integer pageSize) {
```